src/figuresClasses.ts

```diff
@@ -1,19 +1,79 @@
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
+
 export interface Figure {
+  shape: Shape;
+  color: Color;
 
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
 
-}
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (this.a <= 0
+      || this.b <= 0
+      || this.c <= 0) {
+      throw new Error('Invalid value.');
+    }
+
+    const largestSide: number = Math.max(this.a, this.b, this.c);
+
+    if ((this.a + this.b + this.c - largestSide) <= largestSide) {
+      throw new Error('Invalid sizes length.');
+    }
+  }
 
-export class Circle {
+  getArea(): number {
+    const p: number = (this.a + this.b + this.c) / 2;
```

Comment:
Better use a more meaningful variable name instead of `p` - `semiPerimetr` for example.